server/plugins/theme-assets/theme-assets.module.js

```diff
@@ -31,9 +32,9 @@ module.exports.register.attributes = {
  * @return {number}
  */
 internals.getVariationIndex = fileName => {
-    const match = fileName.match(/.+-(.+?)$/);
+    const match = fileName.match(/.+-([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})$/);
```

Comment:
Why don't you use a variable name? seems more descriptive:

```js
fileName.match(new RegExp(`.+-(${uuidRegex})$`);
```

And you could reuse it for the next one:

```js
const match = fileName.match(new RegExp(`(.+)-${uuidRegex}$`));
```